mqttwarn/services/pushsafer.py

```diff
@@ -36,82 +50,217 @@ def pushsafer(**kwargs):
+        try:
+            self.private_key = addrs[0]
+        except IndexError:
+            raise PushsaferConfigurationError(f"Pushsafer private or alias key not configured")
+
+        params = {
+            'expire': 3600,
         }
 
-    if len(addrs) > 1:
-        params['d'] = addrs[1]
+        if len(addrs) > 1:
+            params['d'] = addrs[1]
 
-    if len(addrs) > 2:
-        params['i'] = addrs[2]
+        if len(addrs) > 2:
+            params['i'] = addrs[2]
 
-    if len(addrs) > 3:
-        params['s'] = addrs[3]
+        if len(addrs) > 3:
+            params['s'] = addrs[3]
 
-    if len(addrs) > 4:
-        params['v'] = addrs[4]
+        if len(addrs) > 4:
+            # Special handling for `vibration`.
+            # With the v1 configuration layout, it should not always trigger a vibration when left empty.
+            # Therefore, a synthetic value `0` can be used to configure "device-default" vibration.
+            value = str(addrs[4])
+            if value != "":
+                params['v'] = value
 
-    if len(addrs) > 5:
-        params['u'] = addrs[5]
+        if len(addrs) > 5:
+            params['u'] = addrs[5]
 
-    if len(addrs) > 6:
-        params['ut'] = addrs[6]
+        if len(addrs) > 6:
+            params['ut'] = addrs[6]
 
-    if len(addrs) > 7:
-        params['l'] = addrs[7]
+        if len(addrs) > 7:
+            params['l'] = addrs[7]
 
-    if len(addrs) > 8:
-        params['pr'] = addrs[8]
+        if len(addrs) > 8:
+            params['pr'] = addrs[8]
 
-    if len(addrs) > 9:
-        params['re'] = addrs[9]
+        if len(addrs) > 9:
+            params['re'] = addrs[9]
 
-    if len(addrs) > 10:
-        params['ex'] = addrs[10]
+        if len(addrs) > 10:
+            params['ex'] = addrs[10]
 
-    if len(addrs) > 11:
-        params['a'] = addrs[11]
+        if len(addrs) > 11:
+            params['a'] = addrs[11]
 
-    if title is not None:
-        params['t'] = title
+        if title is not None:
+            params['t'] = title
 
-    try:
-        srv.logging.debug("Sending pushsafer notification to %s [%s]..." % (item.target, params))
-        pushsafer(m=message, k=appkey, **params)
-        srv.logging.debug("Successfully sent pushsafer notification")
-    except Exception as e:
-        srv.logging.warn("Error sending pushsafer notification to %s [%s]: %s" % (item.target, params, e))
-        return False
+        self.params = params
 
-    return True
+    def encode_v2(self):
+        """
+        New-style configuration layout with named parameters for Pushsafer.
+        """
+
+        addrs = self.item.addrs
+        title = self.item.title
+
+        # Decode Private or Alias Key.
+        try:
+            self.private_key = addrs["private_key"]
+        except KeyError:
+            raise PushsaferConfigurationError(f"Pushsafer private or alias key not configured")
+
+        params = {
+            'expire': 3600,
+        }
```

Comment:
Just out of curiosity: Do you think it is still fine to set a default expiration value of 3600 seconds? It might be unexpected to the user. On the other hand, it might have had a reason to implement it like that. Can you remember that, @jpmens or @sumnerboy12?

-- https://www.pushsafer.com/en/pushapi_ext#API-EX

Reply:
> It might be unexpected to the user.

I've documented the current default value with 91881285efde0de2.

Reply:
I have no recollection of why that default was chosen. It does seem very long tho.

Reply:
Thanks for your reply, Ben. The documentation currently does not tell anything about a default value. Would omitting that parameter actually mean that notifications would never expire? Maybe this was the exact reason why that default was chosen -- to have at least some kind of limit?

However, again depending on the use case and scenario, I think 24h may also be a reasonable default value. What if you would be on a flight, for let's say eight hours, or have your phone turned off during the night? Would you completely miss the notifications because they would expire before having the chance to actually get delivered?

What do you think? Maybe removing that default value would make more sense to cover those situations, if you agree that 3600 seconds may actually well be too short instead.

Reply:
> Maybe removing that default value would make more sense to cover those situations, if you agree that 3600 seconds may actually well be too short instead.

In this case, mqttwarn would not interfere at all, and let the user choose the intended value independently.

Reply:
expire=3600 makes no sense that the expire value is passed here

```
if len(addrs) > 10:
            params['ex'] = addrs[10]
```

Reply:
@appzer: 3600 is currently the default value, it can be overwritten when using the `expiration` parameter.

Reply:
but the expire parameter also needs the resend parameter to work
https://www.pushsafer.com/en/pushapi_ext#API-EX

Reply:
Oh I see, thank you so much for telling me how the API works. I actually never used Pushsafer, apologies. In this case, it may make even more sense to remove that ambiguous default value, if no objections will be raised. /cc @jpmens

Reply:
a40efe52ae removes this default value. Thank you again.